### PR TITLE
feat: implement GoReleaser

### PR DESCRIPTION
**Pull request by dejanzele**

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+dist: "dist"
+
+gomod:
+  proxy: true
+  env:
+    - GOPROXY=https://proxy.golang.org,direct
+    - GOSUMDB=sum.golang.org
+  gobinary: go
+
+builds:
```
Comment (severinson) [inline .goreleaser.yml:12]: This is missing binaries (e.g., the lookout ingester).

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+      - "{{ .Env.DOCKER_REPO }}/armada:{{ .Major }}.{{ .Minor }}"
+      - "{{ .Env.DOCKER_REPO }}/armada:{{ .Major }}.{{ .Minor }}.{{ .Patch }}"
+      - "{{ .Env.DOCKER_REPO }}/armada:latest"
+    build_flag_templates:
+      - --label=org.opencontainers.image.title=armada
+      - --label=org.opencontainers.image.description="Armada Bundle"
+      - --label=org.opencontainers.image.url=https://hub.docker.com/r/gresearchdev/armada
+      - --label=org.opencontainers.image.source=https://github.com/G-Research/armada
+      - --label=org.opencontainers.image.version={{ .Version }}
+      - --label=org.opencontainers.image.created={{ time "2006-01-02T15:04:05Z07:00" }}
```
Comment (severinson) [inline .goreleaser.yml:126]: Hardcoded created timestamp in 2006?

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+      - --label=org.opencontainers.image.revision={{ .FullCommit }}
+      - --label=org.opencontainers.image.base.name=alpine:3.17.0
+      - --label=org.opencontainers.image.base.digest=c0d488a800e4
+      - --label=org.opencontainers.image.licenses=Apache-2.0
+      - --label=org.opencontainers.image.vendor=G-Research
+    extra_files:
```
Comment (severinson) [inline .goreleaser.yml:132]: Is this right? I'd have expected only 1 config.

File: build_goreleaser/armada/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# Lookout Ingester
+COPY lookoutingester /app/
+COPY config/eventingester/config.yaml /app/config/eventingester/config.yaml
+
+# Event Ingester
+COPY eventingester /app/
+COPY config/eventingester/config.yaml /app/config/eventingester/config.yaml
+
+# Testsuite
+COPY testsuite /app/
```
Comment (severinson) [inline build_goreleaser/armada/Dockerfile:28]: Prob don't need the testsuite. Enough to have the server components.

**Review by severinson — APPROVED** [review]

File: .goreleaser.yml (hunk 2 above)
Comment (richscott) [reply .goreleaser.yml:126]: I'm not positive, but I think it may be a Golang time format string, used for calling `time.Format(string)`.

**Review by richscott — DISMISSED** [review]

File: .goreleaser.yml (hunk 3 above)
Comment (dejanzele) [reply .goreleaser.yml:132]: yep, the bundled, I renamed armada -> server, and use armada for the bundled image

**Review by severinson — APPROVED** [review]

**Pull request merged**